lib/transactions.rb

```diff
@@ -0,0 +1,41 @@
+class Transactions
+
+  attr_reader :list
+
+  def initialize
+    @list = []
+  end
+
+  def store_transaction(amount, balance)
+    self.list.push({date: format_date, amount: amount, balance: balance})
```

Comment:
Try to avoid using plain data structures in OO code.  OO code combines behaviour with state, but plain data structures are just state.